### ISSUE TITLE
Add the ability to update an uniform array in a constants buffer

### ISSUE THREAD

**Issue by astrochili** [defold/defold#6529]
It's [possible](https://forum.defold.com/t/defold-1-2-189-has-been-released/69707/43?) to update a shader uniform array with `go.set()`:
```
go.set(url, "example", vmath.vector4( 1, 1, 1, 1 ), { index=5 } )
```
But there is no way to do it with a constants buffer created by `render.constant_buffer()`.

Would like to do this by this way:
```
self.constants = render.constant_buffer()
self.constants.light_colors[3] = vmath.vector4(1)
```

**Comment by Dragosha** [comment]
small forum disscussion about this: https://forum.defold.com/t/defold-1-2-189-beta/69611/4

**Comment by Jhonnyg** [comment]
Fixed by #6866
